libs/vkd3d/device.c

```diff
@@ -2234,17 +2293,41 @@ static void d3d12_device_destroy(struct d3d12_device *device)
 static ULONG STDMETHODCALLTYPE d3d12_device_Release(d3d12_device_iface *iface)
 {
     struct d3d12_device *device = impl_from_ID3D12Device(iface);
-    ULONG refcount = InterlockedDecrement(&device->refcount);
+    ULONG cur_refcount = device->refcount;
```

Comment:
This needs to be an (acquire?) atomic load.

Reply:
Done.